src/components/themed/FlipInput2.tsx

```diff
@@ -0,0 +1,248 @@
+import { eq } from 'biggystring'
+import React, { useEffect } from 'react'
+import { Platform, TextInput, TouchableWithoutFeedback, View } from 'react-native'
+import Animated, { AnimationCallback, Easing, interpolate, runOnJS, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated'
+
+import { Fontello } from '../../assets/vector'
+import { useHandler } from '../../hooks/useHandler'
+import { formatNumberInput, isValidInput } from '../../locales/intl'
+import s from '../../locales/strings'
+import { useState } from '../../types/reactHooks'
+import { NumericInput } from '../modals/NumericInput'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { EdgeText } from './EdgeText'
+import { ButtonBox } from './ThemedButtons'
+
+export interface FlipInputGetMethodsResponse {
+  setAmounts: (value: string[]) => void
+}
+
+export type FlipInputFieldInfo = {
+  currencyName: string
+
+  // Maximum number of decimals to allow the user to enter. FlipInput will automatically truncate use input to this
+  // number of decimals as the user types.
+  maxEntryDecimals: number
+}
+
+export interface FlipInputProps {
+  onNext?: () => void
+  convertValue: (sourceFieldNum: number, value: string) => Promise<string | undefined>
+  getMethods?: (methods: FlipInputGetMethodsResponse) => void
```

Comment:
I figured this would be the biggest area of debate. While refs are native to React, their syntax is awkward and a regular developer wouldn't understand them without further doc research (I screwed them up myself). The `getMethods` callback is exactly the API used in the FiatPlugins that you reviewed and its the API we all liked for the SafeSlider. It accomplishes exactly the same thing, provides better security (can't call methods that the component doesn't give access to), and needs no additional understanding of React syntax, and is easier to chain deeper into nested child components. With this, haven't found an advantage to refs.